src/main/java/core/basesyntax/strategy/SupplyAction.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+
+public class SupplyAction implements Action {
+
+    @Override
+    public int action(Operation operation) {
+        Integer value = Storage.storage.get(new Fruit(operation.getNameOfObject()));
+        if (value != null) {
+            return operation.getAmount() + value;
+        }
+        return operation.getAmount();
```

Comment:
You should find a value by key inside your storage and modify it. Check all places